Tests/ApolloCodegenTests/ApolloCodegenTests.swift

```diff
@@ -2334,6 +2334,78 @@ class ApolloCodegenTests: XCTestCase {
     // then
     expect(try ApolloCodegen._validate(config: config)).notTo(throwError())
   }
+  
+  func test__validation__selectionSet_typeConflicts_shouldThrowError() throws {
```

Comment:
I'd love an additional test here that tests this for merged fields.

That test would need `values` to be in an inline fragment. So in another test:
schema:
```graphql
      type Query {
        user: User
      }
      type User {
        containers: [ContainerInterface]
      }
      interface ContainerInterface {
        value: Value
      }
      type Container implements ContainerInterface{
        value: Value
        values: [Value]
      }
      type Value {
        propertyA: String!
        propertyB: String!
        propertyC: String!
        propertyD: String!
      }
```
operation:
"""graphql
query ConflictingQuery {
          user {
              containers {
                  value {
                      propertyA
                      propertyB
                      propertyC
                      propertyD
                  }
                 ... on Container {
                    values {
                        propertyA
                        propertyC
                    }
                  }
              }
          }
      }
"""

Remove the line that validates the `merged` fields, this test should fail. Add the line back in, and it should then throw the validation error on the `Data.User.Container.AsContainer` selection set instead of `Data.User.Container`.